Tests/VaporAWSLambdaRuntimeTests/LambdaServerTests.swift

```diff
@@ -0,0 +1,74 @@
+import AWSLambdaTesting
+@testable import AWSLambdaRuntimeCore
+@testable import AWSLambdaEvents
+import Logging
+import NIO
+import Vapor
+@testable import VaporAWSLambdaRuntime
+import XCTest
+
+final class LambdaServerTests: XCTestCase {
+    func testLambdaServer() {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully()) }
+        
+        let app = Application(.development, .shared(eventLoopGroup))
+        defer { XCTAssertNoThrow(app.shutdown()) }
```

Comment:
@gwynne @0xTim I might need your help here. 

This shutdown call does not propagate to my `LambdaServer` implementation, which implements the `Server` protocol. 

I would expect that the `shutdown()` method of the `Server` is called and the implementation waits for the `onShutdown: EventLoopFuture<Void>` that is provided by the `Server`. Since the shutdown method is not called, I can't shutdown my internal `LambdaLifecycle`. The `LambdaLifecycle` checks its state during `deinit` and crashes if it is not shutdown.

That's why this code crashes right now.